packages/block-library/src/template-part/edit/index.js

```diff
@@ -123,6 +166,23 @@ export default function TemplatePartEdit( {
 	const isEntityAvailable = ! isPlaceholder && ! isMissing && isResolved;
 	const TagName = tagName || areaObject.tagName;
 
+	if ( ! canEditTemplatePart && canViewTemplatePart ) {
```

Comment:
What happens if you can't view or edit, is there even a world where you can't view a template part? Aren't these public?